src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Remote/RemoteAutoInsertOptions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Runtime.Serialization;
+using Microsoft.CodeAnalysis.Razor.Formatting;
+using Microsoft.CodeAnalysis.Razor.Settings;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.Razor.Remote;
+
+[DataContract]
+internal readonly record struct RemoteAutoInsertOptions
+{
+    [DataMember(Order = 0)]
+    public bool EnableAutoClosingTags { get; init; } = true;
+
+    [DataMember(Order = 1)]
+    public bool FormatOnType { get; init; } = true;
+
+    [DataMember(Order = 2)]
+    public RazorFormattingOptions FormattingOptions { get; init; } = new RazorFormattingOptions()
+    {
+        InsertSpaces = true,
```

Comment:
As mentioned in my previous comment, part of the benefit of using a nested property was to have only one place for default options. This code now violates that. This should just create a new formatting options using the default constructor, so it's in charge of its own defaults, exactly as I wrote the code snippet in my previous comment.

Reply:
Sorry, missed that, will do.

Reply:
https://github.com/dotnet/razor/pull/10852